### PR TITLE
Reload Player When Props Have Changed

### PR DESCRIPTION
**Pull request by danmakenoise**
Taken from https://github.com/micnews/react-jw-player/pull/16, but added test coverage.

It is reasonable to expect that when the file or playlist prop changes, the video player should reload. This provides that implementation.

### CODE REVIEW

File: src/react-jw-player.jsx [hunk 1]
```diff
@@ -43,6 +44,12 @@ class ReactJWPlayer extends Component {
       existingScript.onload = getCurriedOnLoad(existingScript, this._initialize);
     }
   }
+  shouldComponentUpdate(nextProps) {
+    return shouldComponentUpdate(this.props, nextProps);
+  }
+  componentDidUpdate() {
+    this._initialize();
```
Comment (ellell) [inline src/react-jw-player.jsx]: Is it necessary to run the full initialize method here again? Do we need to re-attach all the event listeners? And are we sure they aren't being applied twice by running it twice?

File: src/helpers/should-component-update.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+function shouldComponentUpdate(prevProps, nextProps) {
```
Comment (ellell) [inline src/helpers/should-component-update.js]: Also, breaking lifecycle methods out like this makes this harder to read imo. Should be easy to see in the main component file what makes the component update or not. Wdyt?

Comment (danmakenoise) [reply src/helpers/should-component-update.js]: @ellell okay, I understand that. I'll shift it around a bit.

**Comment by danmakenoise** [comment]
@ellell check check check it out

**Pull request merged**